client/js/socket-events/connection.js

```diff
@@ -42,6 +42,11 @@ function handleDisconnect(data) {
 	store.commit("currentUserVisibleError", `Waiting to reconnect… (${message})`);
 	updateLoadingMessage();
 
+	if (message === "xhr poll error" && store.state.serverConfiguration.headerAuthEnabled) {
```

Comment:
I think `store.state.serverConfiguration.headerAuthEnabled` can be `true` while `public` is `false`. Which means that this condition is not properly checking whether header auth is enabled.

Reply:
The whole "reload/reauth on 401" system is a mess at the moment. I'll have to redo this bit and try to move everything into the service worker.